hazelcast-jet-core/src/main/java/com/hazelcast/jet/Jet.java

```diff
@@ -125,6 +132,27 @@ static JetClientInstanceImpl getJetClientInstance(HazelcastInstance client) {
         return new JetClientInstanceImpl(((HazelcastClientProxy) client).client);
     }
 
+    /**
+     * Makes sure that the Jet-expected Hazelcast version and the one in classpath match
+     */
+    private static void assertHazelcastVersion() {
+        String hzVersion = BuildInfoProvider.getBuildInfo().getVersion();
+        try (InputStream resource = Jet.class.getClassLoader().getResourceAsStream("jet-runtime.properties")) {
+            Properties p = new Properties();
+            p.load(resource);
+            String jetHzVersion = p.getProperty("jet.hazelcast.version");
+            if (!hzVersion.equals(jetHzVersion)) {
+                throw new JetException("Jet uses Hazelcast version " + jetHzVersion + " however " +
```

Comment:
I think we should replace Hazelcast and hazelcast occurences with Hazelcast IMDG for better separation

Reply:
I'm not sure, as the dependency is called "hazelcast.jar"